test/tools/OpenCover/OpenCover.psm1

```diff
@@ -1,113 +1,187 @@
+    $AssemblyCoverageChange | Add-Member -MemberType NoteProperty -Name BranchDelta -TypeName [double] -Value $null
+    $AssemblyCoverageChange | Add-Member -MemberType NoteProperty -Name Sequence -TypeName [double] -Value $null
+    $AssemblyCoverageChange | Add-Member -MemberType NoteProperty -Name SequenceDelta -TypeName [double] -Value $null
+    
+    $AssemblyCoverageChange.AssemblyName = $r1.AssemblyName
+    $AssemblyCoverageChange.Branch = $r2.Branch
+    $AssemblyCoverageChange.BranchDelta = $r2.Branch - $r1.Branch
+    $AssemblyCoverageChange.Sequence = $r2.Sequence
+    $AssemblyCoverageChange.SequenceDelta = $r2.Sequence - $r1.Sequence
+
+    return $AssemblyCoverageChange
 }
-Class CoverageChange {
-    [CoverageData]$Run1
-    [CoverageData]$Run2
-    [AssemblyCoverageChange[]]$Deltas
-    [double]$Branch
-    [double]$BranchDelta
-    [double]$Sequence
-    [double]$SequenceDelta
-    CoverageChange([CoverageData]$r1,[CoverageData]$r2) {
-        $this.Run1 = $r1
-        $this.Run2 = $r2
-        $this.Branch = $r2.Summary.BranchCoverage
-        $this.Sequence = $r2.Summary.SequenceCoverage
-        $this.BranchDelta = $r2.Summary.BranchCoverage - $r1.Summary.BranchCoverage
-        $this.SequenceDelta = $r2.Summary.SequenceCoverage - $r1.Summary.SequenceCoverage
-        if ( compare-object ($r2.Assembly.AssemblyName|sort-object)  ($r2.Assembly.AssemblyName|sort-object) ) {
-            Write-Warning "Assembly list differs from run1 to run2"
-        }
-        $h = @{}
-        $r1.assembly | % { $h[$_.assemblyname] = @($_) }
-        $r2.assembly | % { $h[$_.assemblyname] += $_ }
-        $this.Deltas = $h.keys | %{ [AssemblyCoverageChange]::new(@($h[$_])[0],@($h[$_])[1]) }
+
+function Get-CoverageData($xmlPath)
+{
+    [xml]$CoverageXml = get-content -readcount 0 $xmlPath
+    if ( $CoverageXml.CoverageSession -eq $null ) { throw "CoverageSession data not found" }        
+
+    $CoverageData = New-Object System.Management.Automation.PSObject
+    $CoverageData | Add-Member -MemberType NoteProperty -Name CoverageSummary -TypeName [PSObject] -Value $null
+    $CoverageData | Add-Member -MemberType NoteProperty -Name Assembly -TypeName [System.Collections.ArrayList] -Value (New-Object "System.Collections.ArrayList")    
+    $CoverageData | Add-Member -MemberType ScriptMethod -Name GetCoverageSummary -Value { return $this.CoverageSummary }
+    $CoverageData | Add-Member -MemberType ScriptMethod -Name GetAssembly -Value { return $this.Assembly }
+            
+    foreach( $module in $CoverageXml.CoverageSession.modules.module|?{$_.skippedDueTo -ne "MissingPdb"}) {
+        $CoverageData.Assembly.Add((Get-AssemblyCoverageData -element $module)) | Out-Null
     }
+
+    $CoverageData.CoverageSummary = (Get-CoverageSummary -element $CoverageXml.CoverageSession.Summary)
+
+    remove-variable CoverageXml
+    [gc]::Collect()
```

Comment:
This object is *very* large - larger than 1GB, which will relegate it to Gen2 storage (ala LOH) which is not collected often - I saw more than 10/15 minutes elapse after assignment to $null and still no clean up,
a comment as to why it's being done explicitly would certainly not go amiss

---
In reply to: [86645728](https://github.com/PowerShell/PowerShell/pull/2585#discussion_r86645728) [](ancestors = 86645728)